File: backend/WMS.Domain/Entities/Announcement.cs
using System.ComponentModel.DataAnnotations;

namespace WMS.Domain.Entities
{
    public class Announcement : BaseEntity
    {
        public int AnnouncementId { get; set; }
        [Required, MaxLength(200)]
        public string Title { get; set; } = string.Empty;
        [Required]
        public string Content { get; set; } = string.Empty;
        public int PostedBy { get; set; }
        public Employee PostedByEmployee { get; set; } = null!;
        public DateTime PostedAt { get; set; } = DateTime.UtcNow;
        public DateTime? ExpiresAt { get; set; }
        public bool IsActive { get; set; } = true;
    }
}

File: frontend/src/app/models/announcement.ts
export interface AnnouncementDto {
  announcementId: number;
  title: string;
  content: string;
  postedByName: string | null;
  postedAt: string;
  expiresAt: string | null;
  isActive: boolean;
}

export interface CreateAnnouncementDto {
  title: string;
  content: string;
  expiresAt?: string;
}

export interface UpdateAnnouncementDto {
  title?: string;
  content?: string;
  expiresAt?: string;
  isActive?: boolean;
}
